xdsl/rewriter.py

```diff
@@ -69,10 +69,10 @@ def erase_op(op: Operation, safe_erase: bool = True):
         If safe_erase is True, check that the operation has no uses.
         Otherwise, replace its uses with ErasedSSAValue.
         """
-        assert op.parent is not None, "Cannot erase an operation that has no parents"
-
-        block = op.parent
-        block.erase_op(op, safe_erase=safe_erase)
+        if (block := op.parent) is not None:
+            block.erase_op(op, safe_erase=safe_erase)
+        else:
+            op.erase(safe_erase=safe_erase)
```

Comment:
Why is it now ok to erase a operation with no parents?

Reply:
I'll move this to a separate PR as well, the idea is that sometimes we detach an operation and then destroy it.